File: Data/DataRow.swift
//
//  DataRow.swift
//  Incrypt
//
//  Created by Abel Gancsos on 9/2/18.
//  Copyright © 2018 Abel Gancsos. All rights reserved.
//

import Foundation

public class DataRow {
    private var rowIndex   : Int = 0;
    private var rowColumns : [DataColumn] = [];
    
    
    /// This is the default constructor
    public init() {
        
    }
    
    
    /// This is the common constructor
    ///
    /// - Parameter columns: Columns
    public init(columns : [DataColumn]) {
        self.rowIndex = self.rowColumns.count + 1;
        self.rowColumns = columns;
    }
    
    
    /// This is the full constructor
    ///
    /// - Parameters:
    ///   - index: Row index
    ///   - columns: Row columns
    public init(index : Int, columns : [DataColumn]) {
        self.rowIndex = index;
        self.rowColumns = columns;
    }
    
    public func getRowIndex() -> Int {
        return self.rowIndex;
    }
    
    public func getColumns() -> [DataColumn] {
        return self.rowColumns;
    }
	
	public func getColumn(name : String) -> DataColumn? {
		for cursor : DataColumn in rowColumns {
			if(cursor.getName() == name) {
				return cursor;
			}
		}
		return nil;
	}
    
    
    /// This method helps build the row by adding a new column
    ///
    /// - Parameter column: Column
    public func addColumn(column : DataColumn) {
		for cursor : DataColumn in rowColumns {
			if(cursor.getName() == column.getName()) {
				return;
			}
		}
        self.rowColumns.append(column);
    }
}

File: jmagent/packages/data/src/com/data/types/DataRow.java
package com.data.types;
import java.util.ArrayList;

public class DataRow {
	private ArrayList<DataColumn> columns;
	
	public DataRow() {
		this.columns = new ArrayList<DataColumn>();
	}

	public DataRow(ArrayList<DataColumn> columns) {
		this.columns = columns;
	}

	public boolean contains(String name) {
		for (DataColumn column : this.columns) {
			if (column.getColumnName().equals(name)) {
				return true;
			}
		}
		return false;
	}

	public void addColumn(DataColumn column) {
		if (!this.contains(column.getColumnName())) {
			this.columns.add(column);
		}
	}

	public DataColumn getColumn(String name) {
		for (DataColumn column : this.columns) {
            if (column.getColumnName().equals(name)) {
                return column;
            }
        }
		return new DataColumn();
	}	

	public ArrayList<DataColumn> getColumns() { return this.columns; }
}	
